Crash on launch

Hi, I installed everything which was needed and it keeps crashing on launch. The buttons on my g810 do random shiny things but after that I have to stop the programm. Here's the log:

[INFO] 10.06.2016 15:36:05: Device, Logitech, was initialized.
[ERROR] 10.06.2016 15:36:05: Corsair device, CUE Exception! ErrorCode: ServerNotFound.
[INFO] 10.06.2016 15:36:05: Device, Corsair, was not initialized.
[INFO] 10.06.2016 15:36:05: No Chroma SDK available.
[ERROR] 10.06.2016 15:36:05: Razer device, Exception! Message:System.Exception: No Chroma SDK available
   bei Aurora.Devices.Razer.RazerDevice.Initialize() in d:\Repositories\Git\Aurora\Project-Aurora\Devices\Razer\RazerDevice.cs:Zeile 52..
[INFO] 10.06.2016 15:36:05: Device, Razer, was not initialized.
10.06.2016 15:36:05: Listening for game integration calls....
[ERROR] 10.06.2016 15:36:05: Fatal Exception caught : System.Collections.Generic.KeyNotFoundException: Der angegebene Schlüssel war nicht im Wörterbuch angegeben.
   bei System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   bei Aurora.Utils.SteamUtils.GetGamePath(Int32 gameId) in d:\Repositories\Git\Aurora\Project-Aurora\Utils\SteamUtils.cs:Zeile 135.
   bei Aurora.Profiles.Dota_2.Control_Dota2.InstallGSI() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 1079.
   bei Aurora.Profiles.Dota_2.Control_Dota2..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 119.
   bei Aurora.ConfigUI..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\ConfigUI.xaml.cs:Zeile 35.
   bei Aurora.Program.Main(String[] args) in d:\Repositories\Git\Aurora\Project-Aurora\Program.cs:Zeile 208..
[ERROR] 10.06.2016 15:36:05: Runtime terminating: True.
[INFO] 10.06.2016 15:36:24: Device, Logitech, was initialized.
[ERROR] 10.06.2016 15:36:24: Corsair device, CUE Exception! ErrorCode: ServerNotFound.
[INFO] 10.06.2016 15:36:24: Device, Corsair, was not initialized.
[INFO] 10.06.2016 15:36:24: No Chroma SDK available.
[ERROR] 10.06.2016 15:36:24: Razer device, Exception! Message:System.Exception: No Chroma SDK available
   bei Aurora.Devices.Razer.RazerDevice.Initialize() in d:\Repositories\Git\Aurora\Project-Aurora\Devices\Razer\RazerDevice.cs:Zeile 52..
[INFO] 10.06.2016 15:36:24: Device, Razer, was not initialized.
10.06.2016 15:36:24: Listening for game integration calls....
[ERROR] 10.06.2016 15:36:25: Fatal Exception caught : System.Collections.Generic.KeyNotFoundException: Der angegebene Schlüssel war nicht im Wörterbuch angegeben.
   bei System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   bei Aurora.Utils.SteamUtils.GetGamePath(Int32 gameId) in d:\Repositories\Git\Aurora\Project-Aurora\Utils\SteamUtils.cs:Zeile 135.
   bei Aurora.Profiles.Dota_2.Control_Dota2.InstallGSI() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 1079.
   bei Aurora.Profiles.Dota_2.Control_Dota2..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 119.
   bei Aurora.ConfigUI..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\ConfigUI.xaml.cs:Zeile 35.
   bei Aurora.Program.Main(String[] args) in d:\Repositories\Git\Aurora\Project-Aurora\Program.cs:Zeile 208..
[ERROR] 10.06.2016 15:36:25: Runtime terminating: True.
[INFO] 10.06.2016 15:37:31: Device, Logitech, was initialized.
[ERROR] 10.06.2016 15:37:31: Corsair device, CUE Exception! ErrorCode: ServerNotFound.
[INFO] 10.06.2016 15:37:31: Device, Corsair, was not initialized.
[INFO] 10.06.2016 15:37:31: No Chroma SDK available.
[ERROR] 10.06.2016 15:37:31: Razer device, Exception! Message:System.Exception: No Chroma SDK available
   bei Aurora.Devices.Razer.RazerDevice.Initialize() in d:\Repositories\Git\Aurora\Project-Aurora\Devices\Razer\RazerDevice.cs:Zeile 52..
[INFO] 10.06.2016 15:37:31: Device, Razer, was not initialized.
10.06.2016 15:37:31: Listening for game integration calls....
[ERROR] 10.06.2016 15:37:31: Fatal Exception caught : System.Collections.Generic.KeyNotFoundException: Der angegebene Schlüssel war nicht im Wörterbuch angegeben.
   bei System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   bei Aurora.Utils.SteamUtils.GetGamePath(Int32 gameId) in d:\Repositories\Git\Aurora\Project-Aurora\Utils\SteamUtils.cs:Zeile 135.
   bei Aurora.Profiles.Dota_2.Control_Dota2.InstallGSI() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 1079.
   bei Aurora.Profiles.Dota_2.Control_Dota2..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 119.
   bei Aurora.ConfigUI..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\ConfigUI.xaml.cs:Zeile 35.
   bei Aurora.Program.Main(String[] args) in d:\Repositories\Git\Aurora\Project-Aurora\Program.cs:Zeile 208..
[ERROR] 10.06.2016 15:37:31: Runtime terminating: True.
[INFO] 10.06.2016 15:42:50: Device, Logitech, was initialized.
[ERROR] 10.06.2016 15:42:50: Corsair device, CUE Exception! ErrorCode: ServerNotFound.
[INFO] 10.06.2016 15:42:50: Device, Corsair, was not initialized.
[INFO] 10.06.2016 15:42:50: No Chroma SDK available.
[ERROR] 10.06.2016 15:42:50: Razer device, Exception! Message:System.Exception: No Chroma SDK available
   bei Aurora.Devices.Razer.RazerDevice.Initialize() in d:\Repositories\Git\Aurora\Project-Aurora\Devices\Razer\RazerDevice.cs:Zeile 52..
[INFO] 10.06.2016 15:42:50: Device, Razer, was not initialized.
10.06.2016 15:42:51: Listening for game integration calls....
[ERROR] 10.06.2016 15:42:51: Fatal Exception caught : System.Collections.Generic.KeyNotFoundException: Der angegebene Schlüssel war nicht im Wörterbuch angegeben.
   bei System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   bei Aurora.Utils.SteamUtils.GetGamePath(Int32 gameId) in d:\Repositories\Git\Aurora\Project-Aurora\Utils\SteamUtils.cs:Zeile 135.
   bei Aurora.Profiles.Dota_2.Control_Dota2.InstallGSI() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 1079.
   bei Aurora.Profiles.Dota_2.Control_Dota2..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\Profiles\Dota 2\Control_Dota2.xaml.cs:Zeile 119.
   bei Aurora.ConfigUI..ctor() in d:\Repositories\Git\Aurora\Project-Aurora\ConfigUI.xaml.cs:Zeile 35.
   bei Aurora.Program.Main(String[] args) in d:\Repositories\Git\Aurora\Project-Aurora\Program.cs:Zeile 208..
[ERROR] 10.06.2016 15:42:51: Runtime terminating: True.

--- a/Project-Aurora/Project-Aurora/Settings/Layers/Control_EqualizerLayer.xaml
+++ b/Project-Aurora/Project-Aurora/Settings/Layers/Control_EqualizerLayer.xaml
@@ -35,7 +35,7 @@
         </ResourceDictionary>
     </UserControl.Resources>
 
-    <Grid>
+    <Grid Height="255">
         <TextBlock HorizontalAlignment="Left" TextWrapping="Wrap" Text="Equalizer Effect Type: " VerticalAlignment="Top"/>
         <ComboBox x:Name="eq_type" HorizontalAlignment="Left" Margin="120,-2,0,0" VerticalAlignment="Top" Width="177" ItemsSource="{Binding Source={StaticResource EqualizerType}}" ItemTemplate="{Binding Source={StaticResource EqualizerTypeTemplate}}" SelectionChanged="eq_type_SelectionChanged"/>
         <TextBlock HorizontalAlignment="Left" TextWrapping="Wrap" Text="View Type: " VerticalAlignment="Top" Margin="0,27,0,0"/>
@@ -54,11 +54,12 @@
 
         <TextBlock HorizontalAlignment="Left" Margin="302,4,0,0" TextWrapping="Wrap" Text="Max amplitude:" VerticalAlignment="Top"/>
         <xctk:IntegerUpDown x:Name="updown_max_amplitude_value" HorizontalAlignment="Left" Height="24" Margin="389,0,0,0" VerticalAlignment="Top" Width="50" Value="1" Maximum="120" Minimum="1" MouseWheelActiveOnFocus="True" ValueChanged="updown_max_amplitude_value_ValueChanged"/>
-        <CheckBox x:Name="chkbox_dimbgonsound" Content="Dim background on sound" HorizontalAlignment="Left" Margin="302,27,0,0" VerticalAlignment="Top" Checked="chkbox_dimbgonsound_Checked" Unchecked="chkbox_dimbgonsound_Checked"/>
-        <TextBlock HorizontalAlignment="Left" Margin="302,50,0,0" TextWrapping="Wrap" Text="Dim Color:" VerticalAlignment="Top"/>
-        <xctk:ColorPicker x:Name="Clr_dim_color" Margin="363,48,0,0" Height="24" VerticalAlignment="Top" HorizontalAlignment="Left" Width="174" ColorMode="ColorCanvas" SelectedColorChanged="Clr_dim_color_SelectedColorChanged" UsingAlphaChannel="True" />
-        <TextBlock HorizontalAlignment="Left" Margin="302,77,0,0" TextWrapping="Wrap" Text="Frequencies" VerticalAlignment="Top"/>
-        <ListView x:Name="lstbx_frequencies" HorizontalAlignment="Left" Margin="302,98,0,0" Width="100">
+        <CheckBox x:Name="chkbox_dimbgonsound" Content="Dim background on sound" HorizontalAlignment="Left" Margin="302,46,0,0" VerticalAlignment="Top" Checked="chkbox_dimbgonsound_Checked" Unchecked="chkbox_dimbgonsound_Checked"/>
+        <CheckBox x:Name="chkbox_scale_with_system_volume" Content="Scale with system volume" HorizontalAlignment="Left" Margin="302,27,0,0" VerticalAlignment="Top" Checked="chkbox_scale_with_system_sound_Checked" Unchecked="chkbox_scale_with_system_sound_Checked"/>
+        <TextBlock HorizontalAlignment="Left" Margin="303,69,0,0" TextWrapping="Wrap" Text="Dim Color:" VerticalAlignment="Top"/>
+        <xctk:ColorPicker x:Name="Clr_dim_color" Margin="364,67,0,0" Height="24" VerticalAlignment="Top" HorizontalAlignment="Left" Width="174" ColorMode="ColorCanvas" SelectedColorChanged="Clr_dim_color_SelectedColorChanged" UsingAlphaChannel="True" />
+        <TextBlock HorizontalAlignment="Left" Margin="302,89,0,0" TextWrapping="Wrap" Text="Frequencies" VerticalAlignment="Top"/>
+        <ListView x:Name="lstbx_frequencies" HorizontalAlignment="Left" Margin="302,110,0,0" Width="100">
             <ListBox.ItemTemplate>
                 <HierarchicalDataTemplate>
                     <DockPanel HorizontalAlignment="Stretch" >
@@ -68,11 +69,11 @@
                 </HierarchicalDataTemplate>
             </ListBox.ItemTemplate>
         </ListView>
-        <TextBox x:Name="txtBox_newFreqValue" HorizontalAlignment="Left" Height="23" Margin="407,100,0,0" TextWrapping="Wrap" VerticalAlignment="Top" Width="60"/>
-        <TextBlock HorizontalAlignment="Left" Margin="472,104,0,0" TextWrapping="Wrap" Text="Hz" VerticalAlignment="Top"/>
-        <Button x:Name="btn_AddFreq" Content="Add Frequency" HorizontalAlignment="Left" Margin="407,128,0,0" VerticalAlignment="Top" Width="100" Click="btn_AddFreq_Click"/>
-        <Button x:Name="btn_DeleteFreq" Content="Delete Frequency" HorizontalAlignment="Left" Margin="407,155,0,0" VerticalAlignment="Top" Width="100" Click="btn_DeleteFreq_Click"/>
-        <TextBlock HorizontalAlignment="Left" Margin="411,182,0,0" TextWrapping="Wrap" Text="Note: There must be at least 2 frequencies to sample between" VerticalAlignment="Top" Height="46" Width="116"/>
+        <TextBox x:Name="txtBox_newFreqValue" HorizontalAlignment="Left" Height="23" Margin="407,110,0,0" TextWrapping="Wrap" VerticalAlignment="Top" Width="60"/>
+        <TextBlock HorizontalAlignment="Left" Margin="472,114,0,0" TextWrapping="Wrap" Text="Hz" VerticalAlignment="Top"/>
+        <Button x:Name="btn_AddFreq" Content="Add Frequency" HorizontalAlignment="Left" Margin="407,138,0,0" VerticalAlignment="Top" Width="100" Click="btn_AddFreq_Click"/>
+        <Button x:Name="btn_DeleteFreq" Content="Delete Frequency" HorizontalAlignment="Left" Margin="407,165,0,0" VerticalAlignment="Top" Width="100" Click="btn_DeleteFreq_Click"/>
+        <TextBlock HorizontalAlignment="Left" Margin="411,192,0,0" TextWrapping="Wrap" Text="Note: There must be at least 2 frequencies to sample between" VerticalAlignment="Top" Height="46" Width="116"/>
         <Button x:Name="btn_ShowPreviewWindow" Content="Show Preview Window" HorizontalAlignment="Left" Margin="462,0,0,0" VerticalAlignment="Top" Click="btn_ShowPreviewWindow_Click"/>
     </Grid>
 </UserControl>

--- a/Project-Aurora/Project-Aurora/Settings/Layers/Control_EqualizerLayer.xaml.cs
+++ b/Project-Aurora/Project-Aurora/Settings/Layers/Control_EqualizerLayer.xaml.cs
@@ -143,6 +143,12 @@ namespace Aurora.Settings.Layers
                 (this.DataContext as EqualizerLayerHandler).Properties._DimBackgroundOnSound = (sender as CheckBox).IsChecked.Value;
         }
 
+        private void chkbox_scale_with_system_sound_Checked(object sender, RoutedEventArgs e)
+        {
+            if (IsLoaded && settingsset && this.DataContext is EqualizerLayerHandler && sender is CheckBox && (sender as CheckBox).IsChecked.HasValue)
+                (this.DataContext as EqualizerLayerHandler).Properties._ScaleWithSystemVolume = (sender as CheckBox).IsChecked.Value;
+        }
+
         private void Clr_dim_color_SelectedColorChanged(object sender, RoutedPropertyChangedEventArgs<Color?> e)
         {
             if (IsLoaded && settingsset && this.DataContext is EqualizerLayerHandler && sender is Xceed.Wpf.Toolkit.ColorPicker && (sender as Xceed.Wpf.Toolkit.ColorPicker).SelectedColor.HasValue)

--- a/Project-Aurora/Project-Aurora/Settings/Layers/EqualizerLayerHandler.cs
+++ b/Project-Aurora/Project-Aurora/Settings/Layers/EqualizerLayerHandler.cs
@@ -80,6 +80,11 @@ namespace Aurora.Settings.Layers
         [JsonIgnore]
         public float MaxAmplitude { get { return Logic._MaxAmplitude ?? _MaxAmplitude ?? 20.0f; } }
 
+        public bool? _ScaleWithSystemVolume { get; set; }
+
+        [JsonIgnore]
+        public bool ScaleWithSystemVolume { get { return Logic._ScaleWithSystemVolume ?? _ScaleWithSystemVolume ?? false; } }
+
         public bool? _DimBackgroundOnSound { get; set; }
 
         [JsonIgnore]
@@ -178,6 +183,13 @@ namespace Aurora.Settings.Layers
         {
             MMDevice current_device = audio_device_enumerator.GetDefaultAudioEndpoint(DataFlow.Render, Role.Multimedia);
 
+            // The system sound as a value between 0.0 and 1.0
+            float system_sound_normalized = current_device.AudioEndpointVolume.MasterVolumeLevelScalar;
+
+            // Scale the Maximum amplitude with the system sound if enabled, so that at 100% volume the max_amp is unchanged.
+            // Replaces all Properties.MaxAmplitude calls with the sclaed value
+            float scaled_max_amplitude = Properties.MaxAmplitude * (Properties.ScaleWithSystemVolume ? system_sound_normalized : 1);
+
             if (default_device == null || default_device.ID != current_device.ID)
             {
                 UpdateAudioCapture();
@@ -225,7 +237,7 @@ namespace Aurora.Settings.Layers
 
                             Brush brush = GetBrush(fft_val, x, Effects.canvas_width);
 
-                            g.DrawLine(new Pen(brush), x, Effects.canvas_height_center, x, Effects.canvas_height_center - fft_val / Properties.MaxAmplitude * 500.0f);
+                            g.DrawLine(new Pen(brush), x, Effects.canvas_height_center, x, Effects.canvas_height_center - fft_val / scaled_max_amplitude * 500.0f);
                         }
                         break;
                     case EqualizerType.Waveform_Bottom:
@@ -235,7 +247,7 @@ namespace Aurora.Settings.Layers
 
                             Brush brush = GetBrush(fft_val, x, Effects.canvas_width);
 
-                            g.DrawLine(new Pen(brush), x, Effects.canvas_height, x, Effects.canvas_height - Math.Abs(fft_val / Properties.MaxAmplitude) * 1000.0f);
+                            g.DrawLine(new Pen(brush), x, Effects.canvas_height, x, Effects.canvas_height - Math.Abs(fft_val / scaled_max_amplitude) * 1000.0f);
                         }
                         break;
                     case EqualizerType.PowerBars:
@@ -282,7 +294,7 @@ namespace Aurora.Settings.Layers
 
                         for (int f_x = 0; f_x < freq_results.Length - 1; f_x++)
                         {
-                            float fft_val = flux_array[f_x] / Properties.MaxAmplitude;
+                            float fft_val = flux_array[f_x] / scaled_max_amplitude;
 
                             fft_val = Math.Min(1.0f, fft_val);
 

